pkg/controllers/applicationset_controller.go

```diff
@@ -520,6 +520,28 @@ func (r *ApplicationSetReconciler) deleteInCluster(ctx context.Context, applicat
 		_, exists := m[app.Name]
 
 		if !exists {
+
+			// Only check if the finalizers need to be removed IF there are finalizers to remove
+			if len(app.Finalizers) > 0 {
```

Comment:
we should verify that the finalizer here is "resources-finalizer.argocd.argoproj.io" and only delete that one

Reply:
Done.